daemon/libnetwork/sandbox_dns_unix_test.go

```diff
@@ -90,3 +95,69 @@ func TestDNSOptions(t *testing.T) {
 	dnsOptionsList = getResolvConfOptions(t, sb2.config.resolvConfPath)
 	assert.Check(t, is.DeepEqual([]string{"ndots:0"}, dnsOptionsList))
 }
+
+func TestNonHostNetDNSRestart(t *testing.T) {
+	c, err := New(context.Background(), config.OptionDataDir(t.TempDir()))
+	assert.NilError(t, err)
+
+	// Step 1: Create initial sandbox (simulating first container start)
+	sb, err := c.NewSandbox(context.Background(), "cnt1")
+	assert.NilError(t, err)
+
+	sb.startResolver(false)
+
+	err = sb.setupDNS()
+	assert.NilError(t, err)
+	err = sb.rebuildDNS()
+	assert.NilError(t, err)
+
+	// Step 2: Simulate cri-dockerd modifying the resolv.conf for a Kubernetes pause container.
+	// This mimics the behavior where external tools (like cri-dockerd) customize DNS
+	// settings for K8s pods, which should be preserved during container restart/unpause.
+	resolvConfPath := sb.config.resolvConfPath
+	modifiedContent := []byte(`nameserver 10.96.0.10
+search default.svc.cluster.local. svc.cluster.local. cluster.local.
+options ndots:5
+`)
```

Comment:
For the same reason, could you remove anything k8s related from here? A plain `nameserver 1.1.1.1` would do the job.